packages/xl-odt-exporter/src/odt/odtExporter.tsx

```diff
@@ -0,0 +1,342 @@
+      NoInfer<I>,
+      NoInfer<S>,
+      React.ReactNode,
+      React.ReactNode,
+      Record<string, string>,
+      React.ReactNode
+    >["mappings"],
+    options?: Partial<ExporterOptions>
+  ) {
+    const defaults = {
+      colors: COLORS_DEFAULT,
+    } satisfies Partial<ExporterOptions>;
+
+    super(schema, mappings, { ...defaults, ...options });
+    this.options = { ...defaults, ...options };
+  }
+
+  protected async loadFonts() {
+    const interFont = await loadFileBuffer(
+      await import("@shared/assets/fonts/inter/Inter_18pt-Regular.ttf")
+    );
+    const geistMonoFont = await loadFileBuffer(
+      await import("@shared/assets/fonts/GeistMono-Regular.ttf")
+    );
+
+    return [
+      {
+        name: "Inter",
+        fileName: "Inter_18pt-Regular.ttf",
+        data: new Blob([interFont], { type: "font/ttf" }),
+      },
+      {
+        name: "Geist Mono",
+        fileName: "GeistMono-Regular.ttf",
+        data: new Blob([geistMonoFont], { type: "font/ttf" }),
+      },
+    ];
+  }
+
+  public transformStyledText(styledText: StyledText<S>): React.ReactNode {
+    const stylesArray = this.mapStyles(styledText.styles);
+    const styles = Object.assign({}, ...stylesArray);
+
+    if (Object.keys(styles).length === 0) {
+      return styledText.text;
+    }
+
+    const styleName = `BN_T${++this.styleCounter}`;
+
+    // Store the complete style element
+    this.automaticStyles.set(
+      styleName,
+      <StyleStyle style:name={styleName} style:family="text">
+        <StyleTextProperties {...styles} />
+      </StyleStyle>
+    );
+
+    return <TextSpan text:style-name={styleName}>{styledText.text}</TextSpan>;
+  }
+
+  public async transformBlocks(
+    blocks: Block<B, I, S>[],
+    nestingLevel = 0
+  ): Promise<React.ReactNode[]> {
+    const ret: React.ReactNode[] = [];
+    let numberedListIndex = 0;
+
+    for (const block of blocks) {
+      if (block.type === "numberedListItem") {
+        numberedListIndex++;
+      } else {
+        numberedListIndex = 0;
+      }
+
+      const children = await this.transformBlocks(
+        block.children,
+        nestingLevel + 1
+      );
+
+      const content = await this.mapBlock(
+        block as any,
+        nestingLevel,
+        numberedListIndex
+      );
+
+      ret.push(content);
+      if (children.length > 0) {
+        ret.push(...children);
+      }
+    }
+
+    return ret;
+  }
+
+  public async toODTDocument(
+    blocks: Block<B, I, S>[],
+    options?: {
+      header?: string | XMLDocument;
+      footer?: string | XMLDocument;
+    }
+  ): Promise<Blob> {
+    const xmlOptionToString = (xmlDocument: string | XMLDocument) => {
+      let stringifiedDoc = "";
+
+      if (typeof xmlDocument === "string") {
+        stringifiedDoc = xmlDocument;
+      } else {
+        const serializer = new XMLSerializer();
+
+        stringifiedDoc = serializer.serializeToString(xmlDocument);
+      }
+
+      return stringifiedDoc.replace(
```

Comment:
Can you:
- add a comment why this is here / what it does
- include where the regex comes from
- check whether it's necessary for both cases? (i.e.: both when a string vs xmldocument are passed). I don't have the context to judge this